buildSrc/src/main/kotlin/com/getstream/sdk/chat/Dependencies.kt

```diff
@@ -53,6 +53,7 @@ private const val MATERIAL_COMPONENTS_VERSION = "1.2.1"
 private const val MARKWON_VERSION = "4.6.0"
 private const val MOCKITO_KOTLIN_VERSION = "2.2.0"
 private const val MOCKITO_VERSION = "3.5.15"
+private const val MOCKK_VERSION = "1.10.2"
```

Comment:
+1 for MockK

Reply:
I'm gonna be a killjoy here: in this form, -1 for MockK.

We have a tendency of adding dependencies on a whim, often ending up with two-three that perform the same task (look at our assertion libraries, or our image loaders). While the ones in tests at least don't degrade the user experience, they still introduce extra complexity, and require devs to jump back and forth between using different libraries as they touch different tests.

We should either:
- Stick to the mocking solution we already use everywhere (Mockito), if it works well enough for us.
- Make a concerted effort of migrating from the old solution to the new.

If we just add this for one test, I guarantee that in 6 months we'll be looking at a test suite that uses three different mocking libraries seemingly randomly.

Reply:
Agree with Marton. And even more MockK lets you to mock bad things like static dependencies or object dependencies. It tends to make us avoiding  of refactoring of such bad code coupling.

Reply:
That does't seem pragmatic, if you want to mock this with Mockito you have to add getter functions for those properties just for the sake of mocking. I don't feel convinced yet Mock is good enough to justify replacing everything, but it get some things like property mocking right.

Reply:
I'm not sure what you mean about property mocking, this works with Mockito:

```kotlin
whenever(channelControllerMock.unfilteredMessages) doReturn channelMessages
```

Reply:
That breaks for me when an error talking about mockito limitations. I'm not quite sure why

Reply:
FYI, PR for removing it was here https://github.com/GetStream/stream-chat-android/pull/872